### DISCUSSION TITLE
How to dynamically load models when using ScenegraphLayer

### DISCUSSION THREAD

**Discussion by xhy279** [visgl/deck.gl#7369] [Q&A]
Hello,

When using ScenegraphLayer, im expecting to get a reference to the iterating object, so I can load different glb files based on certain property on each item that I passed into the data array,  is this possible?  Or how can i achieve that?  Appreciate!
eg: 

```js
new ScenegraphLayer({
   data:  [item1, item2]
   scenegraph: (item)  => { return different url based on item.type for example}
})
```

**Comment by xhy279** [comment]
or I have to create multiple ScenegraphLayers for each different type?  is that a proper way?

**Reply by xhy279** [reply]
Yeah, it works.

**Reply by chrisgervang** [reply]
Yes, that would be one proper way. Across all layers, each data item will only be passed to accessor props. These are always in the form of `*get*Something`. 

Try packaging multiple scenes into one GLTF file and using `getScene` instead if you'd like switch on data. https://deck.gl/docs/api-reference/mesh-layers/scenegraph-layer#getscene

**Reply by Anand2489** [reply]
@ibgreen @Pessimistress Any suggestions to achieve dynamic loading of 3D geometries using ScenegraphLayer or any other layer ? Similar to IconLayer where we can load 2D icons/textures via `getIcon` accessor

**Reply by Pessimistress** [reply]
As of now, you have to create a separate layer for each scene graph. There isn't any performance advantage to support it in a single layer.
